src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -669,6 +669,7 @@ bool SotaUptaneClient::getNewTargets(std::vector<Uptane::Target> *new_targets, u
 bool SotaUptaneClient::downloadImages(const std::vector<Uptane::Target> &targets) {
   // Uptane step 4 - download all the images and verify them against the metadata (for OSTree - pull without
   // deploying)
+  download_futures_.clear();
```

Comment:
Why does it need to be a member variable? It seems to be local to one method.

Or should it actually be fully asynchronous, like the install case (calling the complete event from a thread)?

Reply:
I think you are right. Currently it is used only locally inside this method. And in the future if we would need wait for download finish from somewhere else, we could move it back.